Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -188,6 +190,33 @@ void HTMLInputElement::set_indeterminate(bool value)
     m_indeterminate = value;
 }
 
+// https://html.spec.whatwg.org/multipage/input.html#compiled-pattern-regular-expression
+Optional<GC::Ref<JS::RegExpObject>> HTMLInputElement::compiled_pattern_regular_expression() const
+{
+    // 1. If the element does not have a pattern attribute specified, then return nothing. The element has no compiled pattern regular expression.
+    auto maybe_pattern = get_attribute(HTML::AttributeNames::pattern);
+    if (!maybe_pattern.has_value())
+        return {};
+
+    // 2. Let pattern be the value of the pattern attribute of the element.
+    auto pattern = *maybe_pattern;
+
+    // 3. Let regexpCompletion be RegExpCreate(pattern, "v").
+    auto& vm = this->vm();
+    auto flags = JS::PrimitiveString::create(vm, "v"_string);
```

Comment:
It matches spec a little less closely, but I do wonder if we should just go straight to a non-JS regex object here instead of through javascript here, since it is not observable.

Reply:
My latest push changes the implementation to use a `Regex<ECMA262>` instead of a `JS::RegexObject`.